src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -404,82 +455,82 @@ protected override void PostStop()
-                    {
-                        return new ListensFailure(listen.AddressesPromise, listens.Exception);
-                    }
-                    else
-                    {
-                        return new ListensResult(listen.AddressesPromise, listens.Result);
-                    }
-                }, TaskContinuationOptions.ExecuteSynchronously)
-                    .PipeTo(Self))
-                .With<ListensResult>(listens =>
+                    return new ListensFailure(listen.AddressesPromise, listens.Exception);
+                }
+                else
                 {
-                    _transportMapping = (from mapping in listens.Results
-                                        group mapping by mapping.Item1.Address
-                                            into g
-                                            select new { address = g.Key, transports = g.ToList() }).Select(x =>
-                        {
-                            if (x.transports.Count > 1)
-                            {
-                                throw new RemoteTransportException(
-                                    string.Format("There are more than one transports listening on local address {0}",
-                                        x.address));
-                            }
-                            return new KeyValuePair<Address, AkkaProtocolTransport>(x.address,
-                                x.transports.Head().Item1.ProtocolTransport);
-                        }).ToDictionary(x => x.Key, v => v.Value);
-
-                    //Register a listener to each transport and collect mapping to addresses
-                    var transportsAndAddresses = listens.Results.Select(x =>
-                    {
-                        x.Item2.SetResult(new ActorAssociationEventListener(Self));
-                        return x.Item1;
-                    }).ToList();
+                    return new ListensResult(listen.AddressesPromise, listens.Result);
+                }
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
+                    .PipeTo(Self));
 
-                    listens.AddressesPromise.SetResult(transportsAndAddresses);
-                })
-                .With<ListensFailure>(failure => failure.AddressesPromise.SetException(failure.Cause))
-                // defer the inbound association until we can enter "Accepting" behavior
-                .With<InboundAssociation>(ia => Context.System.Scheduler.ScheduleTellOnce(TimeSpan.FromMilliseconds(10), Self, ia, Self))
-                .With<ManagementCommand>(mc => Sender.Tell(new ManagementCommandAck(status:false)))
-                // transports are all started. Ready to start accepting inbound associations.
-                .With<StartupFinished>(sf => Context.Become(Accepting))
-                .With<ShutdownAndFlush>(sf =>
+            Receive<ListensResult>(listens =>
```

Comment:
this code is not fundamentally different than it was before, but the diff is funky due to refactoring to a `ReceiveActor`.